File: 프로그래머스/0/181930. 주사위 게임 2/주사위 게임 2.py
def solution(a, b, c):
    s = (a,b,c)
    s = set(s)
    if len(s) ==3:
        answer = a + b + c
    elif len(s) == 2:
        answer = (a + b + c) * (a**2 + b**2 + c**2)
    elif len(s) == 1:
        answer = (a + b + c) * (a**2 + b**2 + c**2) * (a**3 + b**3 + c**3)
    return answer

File: 프로그래머스/0/181930. 주사위 게임 2/주사위 게임 2.java
import java.util.Arrays;
import java.util.HashSet;

class Solution {
    public int solution(int a, int b, int c) {
        HashSet<Integer> set = new HashSet<>(Arrays.asList(a, b,c));
        int answer = 0;
        if (set.size() == 3) {
            answer = a + b + c;
        } else if (set.size() == 2) {
            answer = (a + b + c) * (int)(Math.pow(a, 2) + Math.pow(b, 2) + Math.pow(c, 2));
        } else {
            answer = (a + b + c) * (int)(Math.pow(a, 2) + Math.pow(b, 2) + Math.pow(c, 2)) * (int)(Math.pow(a, 3) + Math.pow(b, 3) + Math.pow(c, 3));
        }
        return answer;
    }
}